### ISSUE TITLE
CA2208 - False positive for specific text in ArgumentException parameter

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn-analyzers#7300]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/CA2208---False-positive-for-specific-tex/10641634)._

---
I got CA2208 message, which doesn't make sense. I was able to reproduce it:

![image.png](https://aka.ms/dc/image?name=Bb9c30e593e3d4a32b5da5f6a7a76df0a638490506966595059_20240418-172455-image.png&tid=b9c30e593e3d4a32b5da5f6a7a76df0a638490506966595059)

Source code:

```
namespace ConsoleApp16;

internal class Program
{
    static void Main()
    {
        Test([1, 2, 3]);
    }

    static void Test(byte[] data)
    {
        throw new ArgumentException("data.Length of encrypted data must be completely divisible by 8.");
    }
}
```

The  message only appears if argument of `ArgumentException` constructor starts with `"data.Length"`. If I simply remove "data", then message disappears:

![image.png](https://aka.ms/dc/image?name=B983bb8a09070416b8fd6f616d3f63180638490509238538462_20240418-172843-image.png&tid=983bb8a09070416b8fd6f616d3f63180638490509238538462)

It seems static analyzer checks for parameter name matching starting value of string? Note: `ArgumentException` has different order of parameters in overload, where both text and parameter name are given, as e.g. `ArgumentOutOfRangeException`. I guess there is a bug, which assume it's must be first and text is used to determine whenever it's parameter name to show the message.

 The suggested fix is also broken, it tries to put whole string into `nameof(...)`, that clearly wouldn't work:

![image.png](https://aka.ms/dc/image?name=B32559a8371b14bd4bf3fe30946b1db42638490512214917687_20240418-173341-image.png&tid=32559a8371b14bd4bf3fe30946b1db42638490512214917687)

---
#### Feedback Bot on 4/19/2024, 00:31 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by sinatrocious** [comment]
@CollinAlpert, it was .Net 8.0. But I can't reproduce it anymore either. It seems the issue was fixed (I just installed Version 17.9.6 update for my VS2022 x64). Close the subject please.
